In [61]:
# Prepare the notebook
import pandas as pd
import plotly.express as px
import numpy as np

# Used the 'thousands' argument as the dataset contains commas
bill = pd.read_csv('meralco_bills.csv', thousands=',')
bill = bill.rename(columns={
    'YEAR' : 'year',
    'MONTH' : 'month',
    'KWH CONSUMPTION' : 'kwh_cons',
    'AMOUNT DUE' : 'amt_due',
    'COST PER KWH ' : 'cost_per_kwh'
})
# Combine the year and month columns into a date type column 
bill['date'] = bill['year'].astype(str) + '-' + bill['month']

# Convert datatype into datetime
bill['date'] = pd.to_datetime(bill['date'], format='%Y-%B')

# Check if data type is correct
bill['date'].dtype


dtype('<M8[ns]')

In [ ]:
# Preview the dataset
bill.head(12)

In [ ]:
# Check the data types
bill.info()

### **1. Does my cost per kWh appear to be increasing or decreasing?**

In [83]:
# Visualize the dataset
figure1 = px.line(
    data_frame=bill,
    x = 'date',
    y = 'cost_per_kwh',
    #color='month',
    labels = {
        'date' : 'Month and Year',
        'cost_per_kwh' : 'Cost per kWh'
    },
    #markers = True,
    #text='cost_per_kwh',
    title = 'Monthly cost per kWh from 2022 to 2026'
)

# For manual control on labels
'''
figure1.update_xaxes(
    dtick = 'M2',
    tickformat = '%b - %Y'
)
'''

figure1.show()

In [ ]:
yearly_avg_cost =  bill.groupby('year')['cost_per_kwh'].mean().reset_index()
# TRUNCATE THE VALUES FOR LABELS
yearly_avg_cost['cost_per_kwh'] = yearly_avg_cost['cost_per_kwh'].map(lambda x: np.trunc(x * 100) / 100)

display(yearly_avg_cost)

,year,cost_per_kwh
0,2022,7.47
1,2023,8.45
2,2024,11.64
3,2025,12.93
4,2026,14.24


In [55]:
figure2 = px.line(
    data_frame = yearly_avg_cost,
    x = 'year',
    y = 'cost_per_kwh',
    labels= {
        'year':'Year',
        'cost_per_kwh':'Cost per kWh'
    },
    markers=True,
    text='cost_per_kwh',
    title = 'Yearly Average Cost per kWh from 2022 to 2026'
    )


# Remove the .5 on X axis
figure2.update_xaxes(type = 'category')

# Adjust text position
figure2.update_traces(textposition='bottom center')

figure2.show()

Based on the yearly average of the cost per kWh, the data shows an overall upward trend from 2022 to 2026. The average cost increased from ₱7.47/kWh in 2022 to ₱14.25/kWh in 2026. Although the monthly costs per kWh shows some fluctuations, the yearly average still increased each year. 

### **2. Does my electricity consumption appear to be increasing or decreasing?**

In [85]:
figure3 = px.line(
    data_frame=bill,
    x = 'date',
    y = 'kwh_cons',
    labels={
        'date': 'Month and Year',
        'kwh_cons': 'kWh Consumption'
    },
    title = 'Monthly kWh Consumption from 2022 to 2026'
)

figure3.show()

In [76]:
yearly_avg_kwh = bill.groupby('year')['kwh_cons'].mean().reset_index()
# Truncate the yearly average kWh consumption
yearly_avg_kwh['kwh_cons'] = yearly_avg_kwh['kwh_cons'].map(lambda x: np.trunc(x * 100) / 100)

display(yearly_avg_kwh)

,year,kwh_cons
0,2022,87.25
1,2023,76.63
2,2024,66.25
3,2025,60.75
4,2026,58.57


In [82]:
figure4 = px.line(
    data_frame=yearly_avg_kwh,
    x = 'year',
    y = 'kwh_cons',
    labels = {
        'year' : 'Year',
        'kwh_cons' : 'kWh Consumption'
    },
    markers=True,
    text='kwh_cons',
    title = 'Yearly Average kWh Consumption from 2022 to 2026'
)

figure4.update_xaxes(type = 'category')
figure4.update_traces(textposition = 'bottom center')

figure4.show()

Based on the yearly average kWh consumption, the data shows an overall downward trend from 2022 to 2026. The average consumption decreased from 87.25 kWh in 2022 to 58.57 kWh in 2026. The monthly data shows some fluctuations with the increases in consumption appearing during some of the hotter months. Overall, the yearly average still decreased each year.